File: C++/Organizm.cpp
#include "Organizm.h"


Organizm::Organizm(Swiat& worldRef, int x, int y) :
	swiat(worldRef)
{
	this->polozenie.x = x;
	this->polozenie.y = y;
	this->wiek = 0;
}

void Organizm::starzejSie()
{
	this->wiek++;
}

Organizm::~Organizm()
{
}

char Organizm::getZnak()
{
	return this->znak;
}

int Organizm::getInicjatywa()
{
	return inicjatywa;
}

int Organizm::getWiek()
{
	return wiek;
}

int Organizm::getX()
{
	return this->polozenie.x;
}

int Organizm::getY()
{
	return this->polozenie.y;
}

bool Organizm::getRozmnozylSie() {
	return this->rozmnozylSie;
}

void Organizm::setX(int x)
{
	this->polozenie.x = x;
}

void Organizm::setSila(int p)
{
	this->sila = p;
}

void Organizm::setY(int y)
{
	this->polozenie.y = y;
}

int Organizm::getSila()
{
	return this->sila;
}

void Organizm::setInicjatywaWiek(int inicjatywa, int wiek)
{
	swiat.aktualizujTabliceInicjatywy(this, inicjatywa, wiek);
	this->wiek = wiek;
	this->inicjatywa = inicjatywa;
	return;
}

void Organizm::setRozmnozylSie(bool rozmnozylSie) {
	this->rozmnozylSie = rozmnozylSie;
}

string Organizm::nazwaOrganizmu() {
	string wartosc;
	switch (this->znak)
	{
	case 'A':
	{
		wartosc += "ANTYLOPA ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'j':
	{
		wartosc += "WILCZEJAGODY ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'b':
	{
		wartosc += "BARSZCZ ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'L':
	{
		wartosc += "LIS ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 't':
	{
		wartosc += "TRAWA ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'g':
	{
		wartosc += "GUARANA ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'C':
	{
		wartosc += "CZLOWIEK ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'O':
	{
		wartosc += "OWCA ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'm':
	{
		wartosc += "MLECZ ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'Z':
	{
		wartosc += "ZOLW ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	case 'W':
	{
		wartosc += "WILK ("; wartosc += to_string(this->polozenie.x); wartosc += ","; wartosc += to_string(this->polozenie.y); wartosc += ")";
		break;
	}
	default:
	{
		return "Unable to read znak";
	}
	}
	return wartosc;
}

File: Java/Organizm.java
import java.awt.*;

public class Organizm {

    public enum TypOrganizmu {
        CZLOWIEK,
        WILK,
        OWCA,
        LIS,
        ZOLW,
        ANTYLOPA,
        TRAWA,
        MLECZ,
        GUARANA,
        WILCZE_JAGODY,
        BARSZCZ_SOSNOWSKIEGO;
    }
    protected Swiat swiat;
    protected int x;
    protected int y;
    protected TypOrganizmu znak;
    private Color kolor;
    protected int sila;
    protected int inicjatywa;
    protected int wiek;
    protected boolean rozmnozylSie;

    private boolean czyUmarl;

    public Organizm(Swiat swiat, int x, int y){
        this.swiat=swiat;
        this.x=x;
        this.y=y;
        this.wiek=0;
        this.czyUmarl=false;
        this.rozmnozylSie=false;
    }
    TypOrganizmu getZnak(){
        return znak;
    }
    int getInicjatywa(){
        return inicjatywa;
    }
    void starzejSie(){
        wiek++;
    }

    void setCzyUmarl(boolean value){
        this.czyUmarl=value;
    }

    public Color getKolor() {
        return kolor;
    }

    public void setKolor(Color kolor) {
        this.kolor = kolor;
    }
    public boolean getCzyUmarl() {
        return czyUmarl;
    }
    int getWiek(){
        return wiek;
    }
    int getX(){
        return x;
    }
    int getY(){
        return y;
    }
    int getSila(){
        return sila;
    }
    boolean getRozmnozylSie(){
       return rozmnozylSie;
    }

    public Swiat getSwiat() {
        return swiat;
    }
    void setX(int x){
        this.x=x;
    }
    void setY(int y){
        this.y=y;
    }
    void setSila(int sila){
        this.sila=sila;
    }
    void setInicjatywaWiek(int inicjatywa, int wiek){
        this.inicjatywa=inicjatywa;
        this.wiek=wiek;
    }

    void setRozmnozylSie(boolean rozmnozylSie){
        this.rozmnozylSie=rozmnozylSie;
    }
    String nazwaOrganizmu(){
        return "Mleko";
    }
    boolean akcja()
    {
        return true;
    };
    boolean rozmnoz(int x, int y){return false;}
    void  wykonajRuch(){}
    boolean kolizja(Organizm other)
    {
        return true;
    };
}
